### PR TITLE
virttest.qemu_storage: add support for luks key management

### PR DESCRIPTION
**Pull request by XueqiangWei**
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>
ID: 1945872

### CODE REVIEW

**Comment by XueqiangWei** [comment]
(1/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_specified_index.q35: PASS (15.92 s)
 (2/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.without_index.q35: PASS (15.70 s)
 (3/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_active_keyslot.q35: PASS (9.95 s)
 (4/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_invalid_keyslot.q35: PASS (9.93 s)
 (5/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_index.q35: PASS (18.21 s)
 (6/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_password.q35: PASS (22.61 s)
 (7/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_only_active_keyslot.q35: PASS (9.90 s)

**Comment by XueqiangWei** [comment]
@zhencliu Please help review it. Many thanks.

File: virttest/qemu_storage.py [hunk 1]
```diff
@@ -1516,12 +1527,24 @@ def amend(self, params, cache_mode=None, ignore_status=False):
                    if key.startswith('amend_')]
         if cache_mode:
             cmd_list.append("-t %s" % cache_mode)
+        secret_objects = self._secret_objects
+        if secret_objects:
+            # add a secret object, use for adding and erasing password
+            sec_id = params["secret_id"]
+            sec_data = params["secret_data"]
```
Comment (zhencliu) [inline virttest/qemu_storage.py:1534]: These two params are used for amend only, can we add the amend_ prefix? and handle them before setting 'options', but the current 'amend_' params are used for options only, e.g. -o x=y,..., so @XueqiangWei @YongxueHong any other comments on this ?

Comment (XueqiangWei) [reply virttest/qemu_storage.py:1534]: zhenchao, the amend_prefix is used for options only, so I didn't add the prefix to secret_id and secret_data. If adding the the amend_ prefix(e.g. amend_secret_id, amend_secret_data), as you said we need handle them before setting options and then pop them from params dict (e.g. del params[\"amend_secret_id\"], del params[\"amend_secret_data\"]). 

I think we can use the abbreviation, e.g. amnd_secret_id and amnd_secret_data. @zhencliu  @YongxueHong What's your opinion?

Comment (XueqiangWei) [reply virttest/qemu_storage.py:1534]: @YongxueHong Please help review the above comments. Many thanks.

Comment (XueqiangWei) [reply virttest/qemu_storage.py:1534]: Thank Yongxue and Zhenchao, I will update it.

**Comment by XueqiangWei** [comment]
(1/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_specified_index.q35: PASS (15.86 s)
 (2/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.without_index.q35: PASS (15.67 s)
 (3/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_active_keyslot.q35: PASS (9.91 s)
 (4/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_invalid_keyslot.q35: PASS (9.92 s)
 (5/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_index.q35: PASS (18.05 s)
 (6/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_password.q35: PASS (22.58 s)
 (7/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_only_active_keyslot.q35: PASS (9.91 s)

**Comment by XueqiangWei** [comment]
@zhencliu @YongxueHong Updated, please help review it again. Many thanks.

File: virttest/qemu_storage.py [hunk 2]
```diff
@@ -1516,12 +1527,24 @@ def amend(self, params, cache_mode=None, ignore_status=False):
                    if key.startswith('amend_')]
         if cache_mode:
             cmd_list.append("-t %s" % cache_mode)
+        secret_objects = self._secret_objects
+        if secret_objects:
+            # add a secret object, use for adding and erasing password
+            sec_id = params["amnd_secret_id"]
+            sec_data = params["amnd_secret_data"]
```
Comment (zhencliu) [inline virttest/qemu_storage.py:1534]: How about giving a brief introduction to these two new params?

Comment (XueqiangWei) [reply virttest/qemu_storage.py:1540]: @zhencliu  Added a brief introduction to doc string, many thanks. Please recheck it. Thanks.

**Comment by zhencliu** [comment]
LGTM

**Comment by XueqiangWei** [comment]
@YongxueHong Please help review it again. Many thanks.

**Comment by XueqiangWei** [comment]
(1/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_specified_index.q35: PASS (15.90 s)
 (2/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.without_index.q35: PASS (15.60 s)
 (3/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_active_keyslot.q35: PASS (9.86 s)
 (4/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_invalid_keyslot.q35: PASS (9.84 s)
 (5/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_index.q35: PASS (18.22 s)
 (6/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_password.q35: PASS (22.60 s)
 (7/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_only_active_keyslot.q35: PASS (9.86 s)

**Comment by XueqiangWei** [comment]
@YongxueHong As we discussed before, I update amnd_ to amend_ for secret_id and secret_data. And Skip them when generating options. I don't pop them from params dict, since they may be used in the second amend testing.
Please help review it again. Many thanks.

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**